@copilot New UT failures detected in **op_ut** tests (4 total). Please check these new failures and analyze whether they are caused by the PR changes.

**Important: Do NOT update the plan in the PR description directly. Use reply comments to update the status.**

Job log: https://github.com/intel/torch-xpu-ops/actions/runs/25922682747
Full new failure report: https://github.com/intel/torch-xpu-ops/actions/runs/25922682747/artifacts/7031431435

**New failures:**

| Category | Class name | Test name | Status | Message |
| --- | --- | --- | --- | --- |
| op_ut | third_party.torch-xpu-ops.test.xpu.dynamo.test_misc_xpu.MiscTests | test_const_dict_variable_python_type | failed | AttributeError: '_thread._local' object has no attribute 'current_tx'
| op_ut | third_party.torch-xpu-ops.test.xpu.profiler.test_profiler_xpu.TestProfiler | test_disable_external_correlation | failed | AssertionError: False is not true
| op_ut | third_party.torch-xpu-ops.test.xpu.profiler.test_profiler_xpu.TestProfiler | test_dynamic_toggle | failed | AssertionError: False is not true
| op_ut | third_party.torch-xpu-ops.test.xpu.test_custom_ops_xpu.TestCustomOpAPI | test_no_grad_skips_autograd | failed | AssertionError: Scalars are not equal!

Instructions:
1. Download the full new failure report artifact for complete error messages
2. Analyze whether the failures are caused by PR changes
3. If caused by PR changes, fix those failures on this PR branch directly
4. Add failing test cases with pytest commands in the PR description for the reproducer job to use
5. Use reply comments to report analysis status - do NOT modify the PR description plan directly

--- a/test/xpu/test_scaled_matmul_cuda_xpu.py
+++ b/test/xpu/test_scaled_matmul_cuda_xpu.py
@@ -9,6 +9,7 @@
 # Owner(s): ["module: intel"]
 
 import torch
+from torch.nn.functional import ScalingType
 from torch.testing._internal.common_device_type import instantiate_device_type_tests
 from torch.testing._internal.common_utils import parametrize, run_tests
 
@@ -22,6 +23,7 @@ with XPUPatchForImport(False):
         e4m3_type,
         e5m2_type,
         scaled_mm_wrap,
+        tensor_to_scale_block,
         TestFP8Matmul,
     )
 
@@ -79,11 +81,69 @@ TestFP8Matmul.test_float8_rowwise_scaling_sanity = parametrize(
 )(_xpu_test_float8_rowwise_scaling_sanity)
 
 
-# NOTE: test_scaled_mm_deepseek_error_messages_* is left at upstream's CUDA-12.9 skip:
-# overriding the body works (skip bypassed), but the asserted error pattern
-# (NotImplementedError "DeepSeek...only supported in CUDA for SM90") is CUDA-specific.
-# XPU's scaled_mm raises ValueError("Invalid scaling configuration") for blockwise scales,
-# so the upstream assertRaisesRegex cannot pass on XPU without changing what it asserts.
+# Override test_scaled_mm_deepseek_error_messages: upstream gates with @onlyCUDA and a
+# CUDA-12.9 skip that trivially evaluates true on XPU (_get_torch_cuda_version() returns
+# (0, 0)).  XPU's _scaled_mm_v2 raises ValueError("Invalid scaling configuration...")
+# for unsupported blockwise scales, so we add an XPU branch alongside the existing
+# CUDA / ROCm ones.
+@parametrize("output_dtype", [torch.bfloat16])
+@parametrize("lhs_block,rhs_block", [(1, 1), (128, 1), (1, 128)])
+@parametrize("M,N,K", [(256, 256, 256), (256, 256, 512)])
+def _xpu_test_scaled_mm_deepseek_error_messages(
+    self, output_dtype, lhs_block, rhs_block, M, N, K, device
+):
+    torch.manual_seed(42)
+
+    x = torch.randn(M, K, device=device, dtype=output_dtype).pow(3)
+    y = torch.randn(N, K, device=device, dtype=output_dtype).pow(3)
+
+    x_fp8, x_scales = tensor_to_scale_block(x, e4m3_type, lhs_block, 128)
+    y_fp8, y_scales = tensor_to_scale_block(y, e4m3_type, rhs_block, 128)
+
+    # 1x128 blocks need scales to be outer-dim-major
+    if lhs_block == 1:
+        x_scales = x_scales.t().contiguous().t()
+        lhs_recipe = ScalingType.BlockWise1x128
+    else:
+        lhs_recipe = ScalingType.BlockWise128x128
+
+    if rhs_block == 1:
+        y_scales = y_scales.t().contiguous().t()
+        rhs_recipe = ScalingType.BlockWise1x128
+    else:
+        rhs_recipe = ScalingType.BlockWise128x128
+
+    # Verify that actual F8 mm raises expected error
+    if torch.version.hip:
+        # ROCm does not yet support DeepSeek-style blockwise scaling
+        expected_error = NotImplementedError
+        expected_pattern = "1x128 and 128x128 scaling not available with ROCm"
+    elif self.device_type == "xpu":
+        # XPU does not support DeepSeek-style blockwise scaling
+        expected_error = ValueError
+        expected_pattern = "Invalid scaling configuration"
+    else:
+        # CUDA non-SM90 should raise NotImplementedError
+        expected_error = NotImplementedError
+        expected_pattern = ".*DeepSeek.*scaling.*only supported in CUDA for SM90.*"
+    with self.assertRaisesRegex(
+        expected_error,
+        expected_pattern,
+    ):
+        scaled_mm_wrap(
+            x_fp8,
+            y_fp8.t(),
+            scale_a=x_scales,
+            scale_recipe_a=lhs_recipe,
+            scale_b=y_scales.t(),
+            scale_recipe_b=rhs_recipe,
+            out_dtype=output_dtype,
+        )
+
+
+TestFP8Matmul.test_scaled_mm_deepseek_error_messages = (
+    _xpu_test_scaled_mm_deepseek_error_messages
+)
 
 
 instantiate_device_type_tests(TestFP8Matmul, globals(), only_for="xpu", allow_xpu=True)

